Windows support - SIMD headers and code change

ashleyabraham opened the pull request:
Added SIMD support to Windows10 x64, MSVC /arch:AVX2, /openmp:llvm

ashleyabraham committed:
Update README.md

Adding support for Windows10 x64 /arch:AVX2 SIMD Intrinsic

ashleyabraham committed:
Added support for Windows intrinsics and added/changed compiler flags for MSVC

ashleyabraham committed:
Update README.md

added notes on windows

ashleyabraham committed:
Update build.yml

adding windows

ashleyabraham committed:
Update README.md

ashleyabraham committed:
Update README.md

Updating examples

ashleyabraham committed:
Update README.md

matsui528 commented:
Hi @ashleyabraham,
Thank you for your PR! First of all, can we make sure that msvc is used in github actions instead of gcc/clang? I'm a bit confused that `CC=gcc/clang` is set even for windows-latest.

ashleyabraham committed:
Update build.yml

working on adding windows without clang

ashleyabraham committed:
Update build.yml

excluding clang for windows

ashleyabraham commented:
I'll be honest, I am not sure why it worked for windows with CC=gcc/clang, github actions somehow calls `C:\Program Files (x86)\Microsoft Visual Studio 14.0\VC\BIN\x86_amd64\cl.exe` but it is able to compile properly. I am not familiar enough to mess with it.

ashleyabraham committed:
Update README.md

updated readme

ashleyabraham committed:
Update README.md

ashleyabraham committed:
Update build.yml

Excluding clang from windows and ubuntu, cl from ubuntu and mac, and gcc from windows

matsui528 commented:
Sorry for the delay in reviewing this. LGTM!
